core/src/main/java/org/apache/accumulo/core/iterators/SortedKeyValueIterator.java

```diff
@@ -147,4 +147,35 @@
    *              if not supported.
    */
   SortedKeyValueIterator<K,V> deepCopy(IteratorEnvironment env);
+
+  /**
+   * Called when the client (i.e. calling iterator or the tserver) will support this iterators ability to yield. If called (and true is returned), then when
+   * next or seek is called this iterator may yield before finding the next top value and key. When this happens, the yield position (returned by
+   * getYieldPosition) will be used when this iterator is rebuilt to seek to the position at which it yielded. This iterator can only yield if the user of this
+   * iterator has called this method. Note that setYieldSupported will never be called in row isolation mode.
+   *
+   * @return <tt>boolean</tt> True if this iterator supported yielding, false otherwise.
+   */
+  default boolean setYieldSupported() {
```

Comment:
suggest having a getter and setter instead of using setYieldSupported as both a setter and getter.